lib/models/recs.js

```diff
@@ -0,0 +1,154 @@
+const User = require('../models/User');
+const fetch = require('node-fetch');
+
+function calculateTopThree(dataArray1, dataArray2) {
+  const dataArray1Reduced = dataArray1.reduce(function (array, element) {
+    return array.concat(element);
+  });
+  const dataArray2Reduced = dataArray2.reduce(function (array, element) {
+    return array.concat(element);
+  });
+
+  const joinedArrays = dataArray1Reduced.concat(dataArray2Reduced);
+
+  const counts = {};
+  joinedArrays.forEach(function (x) {
+    counts[x] = (counts[x] || 0) + 1;
+  });
+
+  const keysSortedDesc = Object.keys(counts).sort(function (a, b) {
+    return counts[b] - counts[a];
+  });
+
+  const topThreeKeys = [
+    keysSortedDesc[0],
+    keysSortedDesc[1],
+    keysSortedDesc[2],
+  ];
+  return topThreeKeys;
+}
+
+async function getMostWatchedGenres(userId) {
+  const user = await User.findById(userId);
+  if (!user) {
+    return 'User not found';
+  } else if (!user.towatch || !user.watched) {
+    return null;
+  } else {
+    const toWatchListAllGenres = user.towatch.map((show) => {
+      const genres = show.genres_id.map((genre) => {
+        return genre;
+      });
```

Comment:
This `.map` doesn't make sense 😃. Input and output are equal.

Reply:
yeah this was a leftover from a different approach before. Thanks for the heads up 👍